### ISSUE TITLE
partial failures in "Unequip Costume" / "Unequip Battle Gear" / "Unequip Pet, Mount, Background"  - sync errors?

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#8781]
From the Report a Bug guild by Lilith of Alfheim (cfeac000-285a-4db4-a624-8412f24484f1):

**I've been recently encountering a bug whenever I use the "Unequip Costume" and "Unequip Pet, Mount, Background" buttons. The latter button works properly about 80% of the time and does absolutely nothing the other 20%. The Costume button on the other hand removes a random selection of items. Sometimes it removes everything but then a couple items pop back seconds later. I refresh the page following the items reappearing and they did remain so it's not a display problem. Manually removing the items by clicking them in the lists individually does not replicate the issue and I've been repeating this for a few weeks to ensure it's not something on my end. I tested this on my mobile tablet through the Chrome web browser and encounter the same problems. It also does not appear to occur after any specific set of actions.**

**I'm on Chrome Version 58.0.3029.110 (64-bit)**
**OS Windows 10 Pro 64-bit**
**UUID cfeac000-285a-4db4-a624-8412f24484f1**

My guess at what might be happening:

I suspect it's a sync error that Habitica is doing to itself. When you click those buttons, exactly the same actions are performed in the back-end as if you'd manually removed all the items one by one, but Habitica performs those actions for you and it won't be delaying briefly between each "click" in the same way that a human does, so several "clicks" are happening pretty much simultaneously. That's the kind of behaviour that causes [sync errors](http://habitica.wikia.com/wiki/Sync_Errors) - i.e., not all of the actions are recorded permanently and so some actions are reverted when the website syncs with the server, meaning that some of the unequipped pieces reappear.

**Comment by trvrfrd** [comment]
I am having a similar (maybe the same?) problem with the Unequip Battle Gear/Costume buttons.

When I hit the button, the UI shows that all items were removed from my avatar, and I get an individual notification for each piece of equipment saying it was unequipped. But, if I refresh the page, only one (and exactly one) item has been unequipped.

What seems to be happening is that a separate POST request is fired off for each slot that has an item currently equipped. When a successful response is returned, the JSON contains a description of what I had equipped, with _only the specific slot from that request_ unequipped (or, to be precise, the base item for that slot is equipped, e.g. armor_base_0). The last request to return successfully always seems to be the one item that is actually unequipped when I refresh the page.

Interestingly, if I just manually unequip all of my items as fast as possible, sometimes more than one item will be successfully unequipped when I refresh the page, but sometimes not.

EDIT: having actually read the wiki post on sync errors linked in the issue, it definitely sounds like a sync error.

Anyway, just wanted to document my findings on this. Let me know if I can provide any further information/investigation!

For reference, I'm on:
- **UUID**: 7620c72a-bb5a-4798-b3c0-846dacb2b44d
- **Browser**: Firefox 55.0.2 64-bit for Ubuntu canonical - 1.0
- **OS**: Xubuntu Linux 16.04.3, kernel 4.4.0-92
